lib/src/main/kotlin/com/swmansion/starknet/service/http/HttpRequest.kt

```diff
@@ -1,28 +1,24 @@
 package com.swmansion.starknet.service.http
 
 import com.swmansion.starknet.provider.Request
-import kotlinx.serialization.DeserializationStrategy
-import kotlinx.serialization.json.Json
 import java.util.concurrent.CompletableFuture
 
+internal typealias HttpResponseDeserializer<T> = (HttpResponse) -> T
```

Comment:
Let's add a comment describing why status code is important here. After gateway is removed we could simplify this.